examples/mi-mctp-ae.c

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+/**
+ * This file is part of libnvme.
+ */
+
+/**
+ * mi-mctp-ae: open a MI connection over MCTP, supporting asynchronous event messages
+ */
+
+#include <assert.h>
+#include <ctype.h>
+#include <err.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stddef.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h> // for usleep
+
+#include <libnvme-mi.h>
+#include <poll.h>
+
+#include <ccan/array_size/array_size.h>
+#include <ccan/endian/endian.h>
+#include <sys/select.h>
+
+// Function to print the byte array
+static void print_byte_array(void *data, size_t len)
+{
+	uint8_t *byte_data = (uint8_t *)data;
+
+	for (size_t i = 0; i < len; ++i)
+		printf("%02X ", byte_data[i]);
+	printf("\n");
+}
+
+static void print_event_info(struct nvme_mi_event *event)
+{
+	printf("aeoi: %02X\n", event->aeoi);
+	printf("aeocidi: %04X\n", event->aeocidi);
+	printf("aessi: %02X\n", event->aessi);
+
+	printf("specific_info: ");
+	if (event->spec_info_len && event->spec_info)
+		print_byte_array(event->spec_info, event->spec_info_len);
+	else
+		printf("EMPTY\n");
```

Comment:
Maybe just skip the `specific_info` message instead of reporting the thing as empty.